cli/src/main/kotlin/commands/UploadResultToSw360Command.kt

```diff
@@ -81,13 +93,55 @@ class UploadResultToSw360Command : CliktCommand(
         val sw360Connection = Sw360Storage.createConnection(sw360Config, sw360JsonMapper)
         val sw360ReleaseClient = sw360Connection.releaseAdapter
         val sw360ProjectClient = sw360Connection.projectAdapter
+        val downloader = Downloader(globalOptionsForSubcommands.config.downloader)
 
         getProjectWithPackages(ortResult).forEach { (project, pkgList) ->
             val linkedReleases = pkgList.mapNotNull { pkg ->
                 val name = createReleaseName(pkg.id)
-                sw360ReleaseClient.getSparseReleaseByNameAndVersion(name, pkg.id.version)
+                val release = sw360ReleaseClient.getSparseReleaseByNameAndVersion(name, pkg.id.version)
                     .flatMap { sw360ReleaseClient.enrichSparseRelease(it) }
                     .orElseGet { createSw360Release(pkg, sw360ReleaseClient) }
+
+                if (attachSources) {
+                    val tempDirectory = createOrtTempDir(pkg.id.toPath())
+                    try {
+                        // First, download the sources of the package into a sources directory, whose root directory is
+                        // temporary.
+                        val sourcesDirectory = tempDirectory.resolve("sources")
+                        val download = downloader.download(pkg, sourcesDirectory)
+
+                        if (!download.matches(pkg)) {
+                            log.debug { "The sources for the package '${pkg.id.toCoordinates()}' could not be " +
+                                    "uploaded to SW360 due to missing source artifact." }
+                        } else {
```

Comment:
I believe we can safely skip this check: If the download fails, a `DownloadException` is thrown. And if it succeeds, the provenance is guaranteed to match the `pkg`. Looking at the provenance is mostly relevant for calls that need to understand from *where* the source code has been downloaded. But in this case we do not care, as long as we *have* the source.